lib/manager/terraform/extract.ts

```diff
@@ -2,83 +2,138 @@ import { logger } from '../../logger';
 import { isValid, isVersion } from '../../versioning/hashicorp';
 import { PackageDependency, PackageFile } from '../common';
 
+export enum TerraformDependencyTypes {
+  unknown = 'unknown',
+  module = 'module',
+  provider = 'provider',
+}
+
+export function getTerraformDependencyType(
```

Comment:
Isn't this function a little overkill? We only have two types that our regex matches, so it's impossible to match default too, right?

Reply:
IMHO I prefer to use enums for such kind of tasks as it is easier to read and it is a build in fail safe for errors.     
From a performance point of view there should be no difference.